cmd/cmount/mount.go

```diff
@@ -143,17 +165,90 @@ func waitFor(fn func() bool) (ok bool) {
 // report an error when fusermount is called.
 func mount(VFS *vfs.VFS, mountpoint string, opt *mountlib.Options) (<-chan error, func() error, error) {
 	f := VFS.Fs()
-	fs.Debugf(f, "Mounting on %q", mountpoint)
 
-	// Check the mountpoint - in Windows the mountpoint mustn't exist before the mount
-	if runtime.GOOS != "windows" {
+	// Check the mountpoint
+	if runtime.GOOS == "windows" {
```

Comment:
Now that this part has got quite long, it might be worth splitting it into a function in its own file.

We can then have a windows and a non-windows version which will be easier to read and save bloating the non-windows versions with code they don't need.

The logic in here is quite complicated now too, so it might be worth splitting the function up a bit to try to make it easier to follow.